lib/runner/androidEmulator.js

```diff
@@ -0,0 +1,87 @@
+const untildify = require('untildify');
+const {Logger} = require('../utils');
+const path = require('path');
+
+const getSdkRootFromEnv = () => {
+  const androidHome = process.env.ANDROID_HOME;
+
+  if (androidHome) {
+    const androidHomeFinal = untildify(androidHome);
+
+    const androidHomeAbsolute = path.resolve(androidHomeFinal);
+    if (androidHomeFinal !== androidHomeAbsolute) {
+      return androidHomeAbsolute;
+    }
+
+    return androidHomeFinal;
+  }
+
+  let err;
+
+  if (androidHome === undefined) {
+    err = new Error('ANDROID_HOME environment variable is NOT set!');
+  } else {
+    err = new Error(`ANDROID_HOME is set to '${androidHome}' which is NOT a valid path!`);
+  }
+  
+  err.help = [
+    `To setup Android requirements, run: ${Logger.colors.cyan('npx @nightwatch/mobile-helper android')}`,
+    `For Android help, run: ${Logger.colors.cyan('npx @nightwatch/mobile-helper android --help \n')}`
+  ];
+
+  err.stack = false;
+
+  throw err;
+};
+
+module.exports = class MobileServer {
+
+  constructor (AVD) {
+    const {getPlatformName, getBinaryLocation} = require('@nightwatch/mobile-helper');
+
+    this.sdkRoot = getSdkRootFromEnv(),
+    this.platform = getPlatformName();
+    this.binaryLocation = getBinaryLocation;
+    this.avd = AVD || 'nightwatch-android-11';
+    this.emulatorId = '';
+    this.emulatorAlreadyRunning = false;
+  }
+
+  async killEmulator() {
+    const {killEmulatorWithoutWait} = require('@nightwatch/mobile-helper');
+
+    return killEmulatorWithoutWait(this.sdkRoot, this.platform, this.emulatorId);
+  }
+  
+  async launchEmulator() {
+    const {getAlreadyRunningAvd, launchAVD} = require('@nightwatch/mobile-helper');
+
+    try {
+      const emulatorAlreadyRunning = await getAlreadyRunningAvd(this.sdkRoot, this.platform, this.avd);
+
+      this.emulatorId = emulatorAlreadyRunning || await launchAVD(this.sdkRoot, this.platform, this.avd);
+      this.emulatorAlreadyRunning = !!emulatorAlreadyRunning;
```

Comment:
why the double negation?

Reply:
Because `emulatorAlreadyRunning` returns a id but I'm storing `this.emulatorAlreadyRunning` as a boolean

Reply:
No but we are only killing the emulator when it has launched by us.

Reply:
Maybe we can change `emulatorAlreadyRunning` to `alreadyRunningAvdEmulatorId`?

Reply:
@garg3133 Are you only suggesting to change the name?

Reply:
Yes, just the local variable name, not the class instance variable name.

Reply:
👍